internal/storage/writer/talaria/talaria.go

```diff
@@ -0,0 +1,62 @@
+package talaria
+
+import (
+	"context"
+	"sync"
+
+	talaria "github.com/kelindar/talaria/client/golang"
+	"github.com/kelindar/talaria/internal/encoding/key"
+	"github.com/kelindar/talaria/internal/monitor/errors"
+)
+
+// Writer to write to TalariaDB
+type Writer struct {
+	lock     sync.Mutex
+	endpoint string
+	client   *talaria.Client
+}
+
+// New initializes a new Talaria writer.
+func New(endpoint string) (*Writer, error) {
+	talaria, err := talaria.Dial(endpoint)
+	if err != nil {
+		return nil, errors.Internal("talaria: unable to connect", err)
+	}
+
+	return &Writer{
+		client:   talaria,
+		endpoint: endpoint,
+	}, nil
+}
+
+// Write will write the ORC data to Talaria
+func (w *Writer) Write(key key.Key, val []byte) error {
+	err := w.tryConnect()
+	if err != nil {
+		return errors.Internal("talaria: unable to connect", err)
+	}
+
+	if err := w.client.IngestORC(context.Background(), val); err != nil {
```

Comment:
Do we need a timeout here?

Reply:
I followed the code in bigquery sink, which did not add any timeouts.
I suppose letting the user input timeout in the config would be preferable?

Reply:
Should be good, since there's already timeout in the talaria client (circuit breaker)